packages/suite/src/reducers/suite/deviceReducer.ts

```diff
@@ -19,9 +19,24 @@ const merge = (device: AcquiredDevice, upcoming: Partial<AcquiredDevice>): Trezo
         ...upcoming,
         state: device.state,
         instance: device.instance,
+        // Don't override features if upcoming device is locked.
+        // In such case the features are redacted i.e. all fields are `null`.
+        features:
+            upcoming.features && isUnlocked(upcoming.features)
+                ? upcoming.features
+                : device.features,
```

Comment:
i thinking about the case where upcoming device is locked. 
upcoming features will not override current one therefore `features.unlocked` will have value `true` but it should have `false`

Reply:
Good point! 👍 With this PR the device could never become locked. What we probably need here is to only protect *some* of the features. We definitely do not want to protect `unlocked`. Are there any other features that actually *practically* change when the device gets locked?

Looking at the fields of `Features` interface, `session_id` seems (just a pure guess based on the name) like sommething statefull. 🤔

Reply:
yes, this should work